datahub-web-react/src/app/homeV3/modules/hierarchyViewModule/childrenLoader/ChildrenLoader.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+import ChildLoader from '@app/homeV3/modules/hierarchyViewModule/childrenLoader/ChildLoader';
+import { ChildrenLoaderType } from '@app/homeV3/modules/hierarchyViewModule/childrenLoader/types';
+
+interface Props {
+    parentValues: string[];
+    loadChildren: ChildrenLoaderType;
+    loadRelatedEntities?: ChildrenLoaderType;
+}
+
+export default function ChildrenLoader({ parentValues, loadChildren, loadRelatedEntities }: Props) {
+    return (
+        <>
+            {parentValues.map((parentValue) => (
+                <ChildLoader
+                    parentValue={parentValue}
+                    key={parentValue}
+                    loadChildren={loadChildren}
+                    loadRelatedEntities={loadRelatedEntities}
+                />
+            ))}
+        </>
+    );
+}
```

Comment:
it's interesting that this and `ChildLoader` are components that we render in a return method instead of just custom hooks to update data. why did you go with that decision? typically i would imagine that if something is in the return signature then it will render something in the UI (even if it's conditional)

not that i disagree, just curious